eng/common/pipelines/templates/archetype-typespec-emitter.yml

```
@@ -447,3 +447,14 @@ extends:
             parameters:
               artifactName: test_artifacts_$(System.JobName)
               artifactPath: $(Build.ArtifactStagingDirectory)
+
+          - ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+              - task: AzureCLI@2
+                displayName: "Upload Spector Standard Coverage Report"
+                condition: and(ne(variables['Build.Reason'], 'PullRequest'), ne('$(SpectorName)', ''))
```

I would move the Build.Reason to the template conditions on line 451. 

```suggestion
                condition: and(ne('$(SpectorName)', ''), succeeded())
```

---

We don't want to upload coverage when this is running from a pr trigger only on merge to main.  Otherwise coverage will bounce all over the place.

Coverage on this dashboard should reflect current main state https://[azure.github.io/typespec-azure/can-i-use/http/](https://azure.github.io/typespec-azure/can-i-use/http/)

---

Talked offline and moved the condition up.